x/did/keeper/grpc_query.go

```diff
@@ -37,7 +39,19 @@ func (k Keeper) DidDocument(
 	ctx := sdk.UnwrapSDKContext(c)
 	did, found := k.GetDidDocument(ctx, []byte(req.Id))
 	if !found {
-		return nil, status.Error(codes.NotFound, "did not found: QueryDidDocument")
+		// it it isn't a key did, return error
+		if !strings.HasPrefix(req.Id, types.DidKeyPrefix) {
+			return nil, status.Error(codes.NotFound, "did not found: QueryDidDocument")
```

Comment:
MAybe we can give more infor about the error? Like Invalid prefix or something like this.